### PR TITLE
Issue #PS-793 feat: Added UI for mark as dropout and remove learner from cohort functionality and added translations for it

### PR DESCRIPTION
**Pull request by Rushikesh-Sonawane99**
## Description

Please describe the changes in your pull request in few words here.

### Changes

List the changes done to fix a bug or introducing a new feature.

## How to test

Describe the steps required to test the changes proposed in the pull request.

Please consider using the closing keyword if the pull request is proposed to
fix an issue already created in the repository
(https://help.github.com/articles/closing-issues-using-keywords/)

## Summary by CodeRabbit

- **New Features**
  - Added new messages for user actions like unmarking dropouts and confirming learner removal in multiple languages (English, Hindi, Marathi, Oriya).

- **Improvements**
  - Enhanced `BottomDrawer` component to conditionally display dropout reasons.
  - Improved `CohortLearnerList` to initialize user data properly and show a message when no data is found.
  - Enhanced `DropOutModal` with better state management and dynamic dropdown rendering.

- **Fixes**
  - Corrected the passing of `cohortId` prop in the `TeachingCenterDetails` component.
  - Fixed the forgot password link in the login page.

- **Refactor**
  - Renamed variable from `userId` to `membershipId` in API calls for clarity.

### CODE REVIEW

File: src/components/BottomDrawer.tsx [hunk 1]
```diff
@@ -25,13 +26,15 @@ interface BottomDrawerProps {
     name: string;
   }[];
   listItemClick: (event: React.MouseEvent, name: string) => void;
+  isDropout: boolean;
```
Comment (itsvick) [inline src/components/BottomDrawer.tsx:29]: @Rushikesh-Sonawane99 keep this component as generic component, dont mix any feature specific ui here

Comment (Rushikesh-Sonawane99) [reply src/components/BottomDrawer.tsx]: @itsvick Created generic component and updated code

**Review by itsvick — CHANGES_REQUESTED** [review]

File: src/components/CohortLearnerList.tsx [hunk 2]
```diff
@@ -54,21 +71,34 @@ const CohortLearnerList = () => {
...
-            return (
-              <LearnersList
-                key={userData.userId}
-                learnerName={userData.name}
-                enrollmentId={userData.enrollmentNumber}
-                isDropout={userData.memberStatus === 'dropout'}
-              />
-            );
-          })}
+      {userData?.map((userData: any) => {
```
[the remark this answers is not in the document]
Comment (Rushikesh-Sonawane99) [reply src/components/CohortLearnerList.tsx:69]: @itsvick replaced userData with data

File: src/components/DropOutModal.tsx [hunk 3]
```diff
@@ -40,7 +46,19 @@ function DropOutModal({ open, onClose }: DropOutModalProps) {
...
-  const theme = useTheme<any>();
+  const handleSelection = (event: SelectChangeEvent) => {
+    setSelectedReason(event.target.value);
+    setIsDisabled(false)
+  };
+
+  const handleMarkDropout = () => {
+    onClose(true, selectedReason);
+    console.log('Dropout api called')
+    setIsDisabled(true)
```
[the remark this answers is not in the document]
Comment (Rushikesh-Sonawane99) [reply src/components/DropOutModal.tsx:57]: @itsvick Replaced it with isButtonDisabled

File: src/components/DropOutModal.tsx [hunk 4]
```diff
@@ -88,25 +106,23 @@ function DropOutModal({ open, onClose }: DropOutModalProps) {
...
+                    key={reason.value}
+                    value={reason.value}
+                    sx={{
+                      fontSize: '16px',
+                      color: theme.palette.warning['300'],
+                    }}
+                  >
+                    {reason.label
+                      .replace(/_/g, ' ')
+                      .toLowerCase()
```
[the remark this answers is not in the document]
Comment (Rushikesh-Sonawane99) [reply src/components/DropOutModal.tsx:122]: @itsvick if lowercase is not done entire sentence comes in upper case.

File: src/components/LearnersList.tsx [hunk 5]
```diff
@@ -34,14 +43,76 @@ const LearnersList: React.FC<LearnerListProps> = ({learnerName, isDropout, enrol
...
+            fontWeight="600"
+          >
+            {t('COMMON.REASON_FOR_DROPOUT')}
+          </Typography>
+          <Typography
+            variant="h3"
+            color={theme.palette.warning[300]}
+            fontWeight="500"
+          >
+            {t('COMMON.REASON_FOR_DROPOUT')}
```
[the remark this answers is not in the document]
Comment (Rushikesh-Sonawane99) [reply src/components/LearnersList.tsx:107]: @itsvick reason will come from api so mean while added reason with comment below

**Pull request merged**